flowkit/flowkit.go

```diff
@@ -365,6 +365,22 @@ func (f *Flowkit) AddContract(
 		return tx.FlowTransaction().ID(), false, trx.Error
 	}
 
+	d := state.Deployments().ByAccountAndNetwork(account.Name, f.network.Name)
+	if d != nil {
+		d.AddContract(config.ContractDeployment{
+			Name: name,
+		})
+	}
+	state.Contracts().AddOrUpdate(config.Contract{
+		Name:     name,
+		Location: contract.Location,
+	})
+
+	err = state.SaveDefault()
```

Comment:
I'm not sure about this, the problem I see is with all the software that uses flowkit, they won't like if the library starts to write some files to their system, especially so out of their control. I feel the save should be inside the internal as it is specific to CLI.

Reply:
I think having this behind a flag would be nice, I can see where frameworks would want both behaviors.

Reply:
IMHO I think calling the save separately is better than introducing a new function argument (it will make APIs messier). I think flowkit shouldn't be saving the state at any point, I think persisting the state is the responsibility of the consumer.

Reply:
Should we then expose the save function so that the consumer can easily get do it if he did not directly create the flowkit instance?